k2/csrc/fsa_utils.cu

```diff
@@ -143,11 +146,14 @@ static Fsa K2AcceptorFromStream(std::istringstream &is) {
       int32_t dest_state = StringToInt(splits[1]);
       int32_t symbol = StringToInt(splits[2]);
       float score = StringToFloat(splits[3]);
+      states_set.insert(src_state);
+      states_set.insert(dest_state);
       arcs.emplace_back(src_state, dest_state, symbol, score);
     } else if (num_fields == 1u) {
       //   0
       // final_state
-      (void)StringToInt(splits[0]);  // this is a final state
+      int32_t final_state = StringToInt(splits[0]);  // this is a final state
```

Comment:
As we assume that the final state always has the largest state number,
is it possible to use 
```cpp
num_states = final_state + 1;
```
?

Reply:
Yes, I think so, but we should probably check that it is the highest numbered state, because it's an error if not.